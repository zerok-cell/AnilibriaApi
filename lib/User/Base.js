"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BaseInstrumental = void 0;
const axios_1 = __importDefault(require("axios"));
/**
 * @description Basic functionality for library classes. It is not intended for non-library use.
 * @description It should only be used for inheritance, because its methods are only available
 * in the class. Transfer the token via super
 * @param token You can get it using the User instance and its tokenGet getter.
 * @param url The address that this class will work with
 *
 * @see tokenGet
 *
 */
class BaseInstrumental {
    token;
    barerToken;
    urlReferences;
    constructor(token, url) {
        this.token = token;
        this.barerToken = "Bearer " + token;
        this.urlReferences = url;
    }
    /**
     *
     * @param m Method request from axios, example GET,POST,DELETE and many others type request
     * @param br To send or not to send a bearer token. A bearer token is a token derived from the addition string of
     * the bearer and the token instance that you passed to the constructor.
     * @param url The endpoint of the request address.  The request address is derived from the addition of the address
     * that you passed to the constructor and that you passed here.
     * If the address that you passed to the constructor is for example `https://example.com/` and the url parameter is
     * equal to `bar`, then the final request address will be equal to
     * `https://example.com/bar`
     * @param d Data in the form of {key:value}
     * @returns {Promise<Record<any,any>>} Status code(http code) and the data that came from the server
     */
    requestUrl = async (m, br, url, d) => {
        const { status, data } = await axios_1.default.request({
            url: this.urlReferences + url,
            method: m, data: d, headers: {
                "Authorization": br && this.barerToken,
                "Content-Type": "application/json"
            }
        });
        return { status, data };
    };
}
exports.BaseInstrumental = BaseInstrumental;
