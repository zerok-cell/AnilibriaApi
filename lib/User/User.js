"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.User = void 0;
const axios_1 = __importDefault(require("axios"));
const dotenv_1 = __importDefault(require("dotenv"));
// Загружаем переменные окружения из .env файла
dotenv_1.default.config();
class User {
    privateToken;
    login;
    password;
    urlUser;
    constructor(login, password) {
        this.login = login;
        this.password = password;
        this.urlUser = "https://anilibria.top/api/v1/accounts/users/auth/";
    }
    get token() {
        return this.privateToken;
    }
    authorize = async () => {
        const fetchUrl = await axios_1.default.post(`${this.urlUser}login/`, {
            login: this.login,
            password: this.password
        });
        this.privateToken = fetchUrl.data.token;
        return { status: fetchUrl.status, token: this.privateToken };
    };
    deauthorize = async () => {
        if (this.privateToken) {
            try {
                const fetchUrl = await axios_1.default.post(`${this.urlUser}logout/`, {}, {
                    headers: {
                        "Authorization": `Bearer ${this.privateToken}`
                    }
                });
                this.privateToken = undefined;
                return fetchUrl.status;
            }
            catch (error) {
                return error;
            }
        }
        else {
            return "Nothing token";
        }
    };
}
exports.User = User;
// const main = async () => {
//   const user = new User("zerok-cell", "Duplex007");
//
//   try {
//     const result = await user.authorize();
//     console.log(result);
//   } catch (error) {
//     console.error("Ошибка авторизации:", error);
//   }
// };
//
// main();
