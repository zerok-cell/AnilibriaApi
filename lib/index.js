"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const User_1 = require("./User/User");
const main = async () => {
    const user = new User_1.User("zerok-cell", "Duplex007");
    try {
        const result = await user.authorize();
        // const ref = new  References(user.tokenGet)
        // const age = await ref.genRes()
        const deauth = user.logout();
        console.log(user.tokenGet);
    }
    catch (error) {
        console.error("Ошибка авторизации:", error);
    }
};
main();
