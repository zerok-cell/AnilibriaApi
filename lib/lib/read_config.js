"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.configLoad = exports.ConfigSchema = void 0;
const fs_1 = __importDefault(require("fs"));
const js_yaml_1 = __importDefault(require("js-yaml"));
const path_1 = __importDefault(require("path"));
const zod_1 = require("zod");
exports.ConfigSchema = zod_1.z.object({
    app: zod_1.z.object({
        url: zod_1.z.string().nonempty(),
    })
});
const configPath = path_1.default.join(__dirname, "config.yaml");
/**
 * @description Use exclusively inside THE library IS `NOT INTENDED FOR OUTDOOR USE`
 * @return {ConfigType} Returns the config
 */
const configLoad = () => {
    const file = fs_1.default.readFileSync(configPath, "utf8");
    return js_yaml_1.default.load(file);
};
exports.configLoad = configLoad;
