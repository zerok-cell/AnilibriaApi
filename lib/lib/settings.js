"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.anilibria = void 0;
const axios_1 = __importDefault(require("axios"));
exports.anilibria = axios_1.default.create({
    baseURL: "https://anilibria.top/api/v1"
});
