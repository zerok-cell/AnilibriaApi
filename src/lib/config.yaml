app:
  url: "https://anilibria.top/api/v1/"