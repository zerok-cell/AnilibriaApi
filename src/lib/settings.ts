import axios from "axios";

export const anilibria = axios.create({
  baseURL:"https://anilibria.top/api/v1"
})
