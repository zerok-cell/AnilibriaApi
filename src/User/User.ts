import { anilibria } from "../lib/settings";
import { DataAndStatus } from "../zod_scheme/user.schema";


/**
 * @author Murad Shakhsinov
 * @description A class for interacting with endpoints related to the entrance user's account It is mandatory to accept
 * the password and email address of the `anilibria` user.
 * @param {string} [login] User name on anilibria
 * @param {string} [password] Anilibria login password
 * @example const user = User("my_login", "my_password")
 * @version 1.0.0
 * @since 1.0.0
 */
export class User {
  private readonly login: string;
  private readonly password: string;


  constructor(login: string, password: string) {
    this.login = login;
    this.password = password;
  }

  /**
   * @description issues you a private token
   * @example
   * const index = async () => {
   *   const user = User("my_login", "my_password")
   *   const token = user.tokenGet
   * }
   * @return {{ tokenBearer:string,token:string }}
   */
  public get tokenGet(): { tokenBearer: string, token: string } {
    const tokenBearer = anilibria.defaults.headers.common.Authorization as string;
    const token = anilibria.defaults.headers.common.Authorization as string;
    return { tokenBearer, token: token.slice(7) };
  }

  /**
   * @description authorize the user using the login and password data that you passed to the instance of the class
   * @example
   * @example
   * const index = async () =>{
   *   const user = User("my_login", "my_password")
   *   const result = await user.authorize()
   * }
   *  @returns {Promise<DataAndStatus<Record<string, string>>} The operation status (http code)
   *  and the data received as a result of the request
   *  @version 1.0.1
   *
   */
  public authorize = async (): Promise<DataAndStatus<{ token: string }>> => {

    const { status, data } = await anilibria.post<{ token: string }>(`/accounts/users/auth/login`, {
      login: this.login,
      password: this.password

    });
    anilibria.defaults.headers.common.Authorization = "Bearer " + data.token;
    return { data, status };


  };

  /**
   * @description Makes the user log out by `deleting` the token from the instance
   * @description It won't work if there is no token.
   * @returns {Promise<DataAndStatus<Record<string, any>>>} | Status code(http) or `{data:string, status:number}`
   * @throws {Error} The error may be related to the internal structure of the code, how you use this method,
   * or an error in the request.
   * @version 1.0.1
   */
  public logout = async (): Promise<DataAndStatus<Record<string, any>>> => {
    if (anilibria.defaults.headers.common.Authorization) {
      try {
        const { data, status } = await anilibria.post<{ token: null }>(`/accounts/users/auth/logout`);
        anilibria.defaults.headers.common.Authorization = undefined;
        return { data, status };
      } catch (error) {
        throw new Error(`There's been some kind of mistake. - ${error}`);
      }
    } else {
      return { data: "The token was not found in the instance, you may not be registered.", status: 401 };
    }

  };
}
