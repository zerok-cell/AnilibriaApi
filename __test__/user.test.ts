import {
  describe, expect, test
} from "@jest/globals";
import axios from "axios";
import { anilibria } from "../src/lib/settings";
import { User } from "../src/User/User";

const user = new User("zerok-cell", "Duplex007");


test("User auth and logout", async () => {
  const auth = await user.authorize();
  expect(auth.status).toBe(200);
  console.log(auth.data);
  expect(typeof anilibria.defaults.headers.common.Authorization === "string").toBe(true);


  expect(await user.logout()).toStrictEqual({"data": {"token": null}, "status": 200});
  expect(user).toMatchSnapshot();

});

