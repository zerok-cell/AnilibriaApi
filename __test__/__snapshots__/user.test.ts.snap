// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User auth and logout 1`] = `
User {
  "authorize": [Function],
  "login": "zerok-cell",
  "logout": [Function],
  "password": "Duplex007",
}
`;
